src/test/coins_tests.cpp

```diff
@@ -747,25 +754,22 @@ BOOST_AUTO_TEST_CASE(ccoins_spend)
     CheckSpendCoins(VALUE1, VALUE2, ABSENT, DIRTY|FRESH, NO_ENTRY   );
 }
 
-static void CheckAddCoinBase(CAmount base_value, CAmount cache_value, CAmount modify_value, CAmount expected_value, char cache_flags, char expected_flags, bool coinbase)
+static void CheckAddCoinBase(CAmount base_value, CAmount modify_value, const CoinEntry& cache_coin, const CoinEntry& expected, bool coinbase)
 {
-    SingleEntryCacheTest test(base_value, cache_value, cache_flags);
-
-    CAmount result_value;
-    char result_flags;
     try {
+        SingleEntryCacheTest test(base_value, cache_coin);
```

Comment:
I was hesitant at first (mostly because this will be removed in `test: Validate error messages on fail`), but we want to always fail when these do, so you're right, thanks!